backend-shared/config/db/ephemeralPostgres.go

```diff
@@ -0,0 +1,210 @@
+package db
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type EphemeralDB struct {
+	db            *PostgreSQLDatabaseQueries
+	dbContainerID string
+	dockerNetwork string
+}
+
+// NewEphemeralCreateTestFramework will create a docker container,
+// network, connect to database and populate the schema. This is
+// merely defined for testing purpose
+func NewEphemeralCreateTestFramework() (EphemeralDB, error) {
+	dockerName := "managed-gitops-postgres-test"
+	dockerNetworkcmd := "docker network create %s"
+	uuid := uuid.New().String()
+	tempDBName := "db-" + uuid
+	tempNetworkName := "gitops-net-" + uuid
+	s := fmt.Sprintf(dockerNetworkcmd, tempNetworkName)
```

Comment:
Again, from the lower comment I made this is by mistake committed, I have a commit in my local that fixes this, we have a function generate_uuid() already for this. I'll push a commit for this.
I was testing in my personal repository here samyak-jn/go-utilities, and that's where for ease I copied it directly.

Reply:
I'll research about it, but when I tested for these locally the network was getting created and deleting on the name generated by this seamlessly. 

```
➜ samjain@fedora  ~  docker network create gitops-net-dff7485d-25d1-4b0f-aa90-ba5af698dd7d
6683c7c745290d55633fe574ebfa09c97ebd7c15da74dc55c0b4f365f5b76dcb
➜ samjain@fedora  ~  docker network  ls                                                   
NETWORK ID     NAME                                              DRIVER    SCOPE
32dac68dd470   bridge                                            bridge    local
cd6a6008f06c   gitops-net                                        bridge    local
6683c7c74529   gitops-net-dff7485d-25d1-4b0f-aa90-ba5af698dd7d   bridge    local
2888161aca9d   host                                              host      local
25876db9a143   k3d-my-cluster                                    bridge    local
fcb0b2367961   none                                              null      local
➜ samjain@fedora  ~  docker network rm gitops-net-dff7485d-25d1-4b0f-aa90-ba5af698dd7d    
gitops-net-dff7485d-25d1-4b0f-aa90-ba5af698dd7d

```

Reply:
For the extra variable that I created (instead of s, that I removed now that is just used for printing purpose and will be removed) but for others, each tends to hold a meaning, let's say about above example, the same value in `tempNetworkName` is needed in another function as well to clean what's created and hence is separately created, but I'll skim once again to remove unwanted variable use.